runtime/context.go

```diff
@@ -434,7 +434,10 @@ type Logger interface {
 }
 
 // ContextMetrics emit metrics with tags extracted from context.
+// Methods that update metrics create a new tally scope on each call. This is unnecessary slow in scenarios that update
+// multiple metrics.
 type ContextMetrics interface {
+	Scope() tally.Scope
```

Comment:
maybe we can pass ctx in Scope() which would return tagged scope.

Suggesting this as the name is context metrics